ckanext/xloader/jobs.py

```diff
@@ -212,7 +212,11 @@ def direct_load():
         set_datastore_active(data, resource, logger)
         if 'result_url' in input:
             job_dict['status'] = 'running_but_viewable'
-            callback_xloader_hook(result_url=input['result_url'],
+            callback_url = config.get('ckanext.xloader.site_url') or config.get('ckan.site_url')
+            callback_url = urljoin(
+                callback_url.rstrip('/'), '/api/3/action/xloader_hook')
+            result_url = callback_url
```

Comment:
Why create `result_url` at all?

Reply:
You mean assign `callback_url` back to `result_url` ? I'm try to minimise changes to the rest of the XLoader code. If you think I should just use `callback_url` then I can do that

Reply:
You're already changing from `result_url=input['result_url']` to `result_url=result_url`. Why not just change it to `result_url=callback_url`? Or just call it `result_url` instead of `callback_url` in the first place.